### ISSUE TITLE
Produce a source distribution file to pypi (tar.gz format)

### ISSUE THREAD

**Issue by ChrisCarini** [matrix-profile-foundation/matrixprofile#61]
**Is your feature request related to a problem? Please describe.**
In some corporate environments, it's common to import external libraries into an internal repository for internal dependency resolution. In order to import `matrixprofile` for use, a `tar.gz` file is needed similar to the python libraries `requests` or `pytest`.

**Describe the solution you'd like**
A `tar.gz` file uploaded to pypi as part of the release. See `requests` or `pytest` as examples. Links below.

**Describe alternatives you've considered**
None.

**Additional context**
* https://pypi.org/project/requests/#files
* https://pypi.org/project/pytest/#files

**Comment by tylerwmarrs** [comment]
Thank you for raising this issue. The packages distributed on PyPi are currently binary wheels to avoid complications with dependencies during installation. A "tar.gz" deployment is distributing the source code. It is my understanding that some companies prefer to have the "tar.gz" file purely to execute some vulnerability scanning on the code. The "tar.gz" is the older way of distributing a library using "sdist" instead of the newer "wheel" format. See https://www.python.org/dev/peps/pep-0427/#id5

The two packages that you linked can easily distribute source code and installation is simplistic. This is because they are using pure Python while our library also uses Cython for speed reasons. Numpy is in a similar situation as us where the installation may be cumbersome for end-users when distributing source code. However, they do provide their source code as a ".zip".

![image](https://user-images.githubusercontent.com/18449075/100768654-c1050080-33c0-11eb-9319-8c4c2c4362d9.png)

Do you know if a ".zip" is suitable for your employer or is the older sdist (.tar.gz) source required?

**Comment by tylerwmarrs** [comment]
@ChrisCarini are you able to provide more insight?

**Comment by ChrisCarini** [comment]
Hey @tylerwmarrs - apologies, I missed the notification for your first detailed (thank you, by the way! :) ) response.

I'm confirming on the `.zip` vs `.tar.gz` - I _believe_ a `.zip` should be fine, but am waiting on confirmation.

I do know my employer is getting support soon for consuming the wheel format, however, it will require a longer process to bring in, as it will require our company to manually sign off from a legal and security perspective (whereas, this process is simplified / more automated for sdists).

For the time being, I can manually pull this in for experimenting locally - let me get back to you regarding the `.zip` vs `.tar.gz` though!

And, thank you again for the detail above, python packaging is a bit out of my wheelhouse (see what I did there?! ha ha), so this helps me get more familiar with it!

**Comment by tylerwmarrs** [comment]
@ChrisCarini did you determine if the zip file works for your employer? Either way, I think we should probably add the source distribution as described above.

**Comment by ChrisCarini** [comment]
Hi @tylerwmarrs - confirmed with my employer, so long as it is a sdist, we should be able to bring it in for use.

We are working on a way to bring in wheels, although that is not as automated, yet (it requires manual approvals from our security and legal teams).

**Comment by tylerwmarrs** [comment]
@ChrisCarini thank you for figuring this out. I'll update the title to reflect the need for a "zip" file.

**Comment by ChrisCarini** [comment]
Thank you! And happy holidays!

**Comment by tylerwmarrs** [comment]
@ChrisCarini happy holidays!

@lmmentel this is another easy one if you have time. :)

**Comment by lmmentel** [comment]
Sure, it is actually possible to have both ".tar.gz" and ".zip" according to the [docs](https://docs.python.org/3/distutils/sourcedist.html#creating-a-source-distribution). I can add both options to travis build.

**Comment by tylerwmarrs** [comment]
I started working on this myself as I found time to do it. :) It probably makes sense for me to tackle it since I have ownership of the TravisCI etc.

**Comment by tylerwmarrs** [comment]
@lmmentel you actually cannot upload multiple archive types to PyPi. See screenshot:

![image](https://user-images.githubusercontent.com/18449075/103027375-55273b00-451b-11eb-969d-d768ff313e91.png)

**Comment by tylerwmarrs** [comment]
@lmmentel you actually can only upload one archive type to PyPi. See screenshot:
![image](https://user-images.githubusercontent.com/18449075/103027499-991a4000-451b-11eb-97df-4f239b80aede.png)

**Comment by lmmentel** [comment]
I didn't know PyPI had this limitation. Do you want me to update the PR or you want to take it from here?

**Comment by tylerwmarrs** [comment]
I got it. Thanks for your help today! I only have to work a half day today and was going to tackle some of these "easy" issues. It is always nice to have help!

**Comment by lmmentel** [comment]
Glad to be useful. 

I guess I thought that PyPI is more like github releases allowing multiple source formats.

![image](https://user-images.githubusercontent.com/8989838/103028491-2aa7a300-4558-11eb-8e8c-1fd5c78a8480.png)

BTW @ChrisCarini any reason you couldn't use source distributions [released on github](https://github.com/matrix-profile-foundation/matrixprofile/releases/tag/v1.1.5)?

**Comment by ChrisCarini** [comment]
Wow, that was a fast turn around! Thank you!!

@lmmentel - our internal process for python is automated from pypi. Manually bringing in other source releases from GitHub release would require similar manual security / legal review, unfortunately. Having it on pypi makes it much easier both to get started, and when we want to bring in the latest releases.

**Comment by lmmentel** [comment]
Got it, thanks for the explanation.
